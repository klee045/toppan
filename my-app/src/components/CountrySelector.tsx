import React, {
  CSSProperties,
  Dispatch,
  SetStateAction,
  useState,
} from "react";

const CountrySelectorStyle: CSSProperties = {};

const CountrySelector = (): JSX.Element => {
  const [countrySelected, setCountrySelected]: [
    string,
    Dispatch<SetStateAction<string>>
  ] = useState("SG");

  const handleClick = (): void => {
    const countries: string[] = ["SG", "US", "MY"];
    const randomId: number = Math.floor(
      Math.random() * (countries.length - 0 + 1) + 0
    );
    setCountrySelected(countries[randomId]);
  };

  return <button onClick={handleClick}>Get Country: {countrySelected}</button>;
};

export default CountrySelector;
