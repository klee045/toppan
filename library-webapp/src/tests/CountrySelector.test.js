import React from "react";
import renderer from "react-test-renderer";
import CountrySelector from "../components/CountrySelector";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer
    .create(<CountrySelector countrySelected={"SG"} handleClick={""} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('has a div with id "action-btn"', () => {
  const tree = renderer.create(
    <CountrySelector countrySelected={"SG"} handleClick={""} />
  );
  expect(
    shallow(<CountrySelector countrySelected={"SG"} />)
      .find("div#action-btn")
      .exists()
  ).toBe(true);
});
