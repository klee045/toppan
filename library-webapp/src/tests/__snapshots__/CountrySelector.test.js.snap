// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  id="action-btn"
  onClick=""
  style={
    Object {
      "alignSelf": "flex-start",
      "backgroundColor": "rgb(241, 230, 230)",
      "borderColor": "rgb(255, 116, 43)",
      "borderWidth": "1px",
      "fontSize": "18px",
      "height": "5%",
      "marginLeft": "25px",
      "marginTop": "25px",
      "padding": "4px",
      "width": "9%",
    }
  }
>
  Get country: 
  SG
</button>
`;
